### PR TITLE
Restoring db manually and via shell script

### LINKED ISSUE #4: Import Catalogue
First steps should probably be to setup the catalogue.

Programatically restore the GCDB backup into MySQL (http://www.comics.org/download/). Add something somewhere that attributes usage rights for GCDB https://docs.comics.org/wiki/App_Guidelines

- https://docs.comics.org/wiki/Main_Page#Technical_Details
- https://docs.comics.org/wiki/App_Guidelines
- https://github.com/GrandComicsDatabase/gcd-django
- http://www.comics.org/download/

### PR DESCRIPTION
**Pull request by ajbovaird**
This is a first pass, probably a pretty hacky attempt.

Closes #4

### CODE REVIEW

**Review by Jayracine — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -8,3 +8,20 @@ Frontend: React (TypeScript?), Bootstrap
...
+1. Download from https://www.comics.org/download/ (account required), unzip to project root
+1. `just up` to start containers
+1. Open a terminal in the container using `docker exec -ti {mysql_container_id} bash`
+1. Enter the mysql CLI using `mysql -u root -p`
+1. Run the following:
+    * `create schema gcdonline;`
+    * `create user gcdonline;`
+    * `grant all on gcdonline.* to gcdonline;`
+1. Type `exit` to close the mysql cli, then `exit` again to exit the container shell
+1. From project root, run `docker exec -i {mysql_container_id} /usr/bin/mysql -u root --password={password} gcdonline < {extracted_filename}.sql` (takes a bit of time)
```
Comment (EduardoPerezExt) [inline README.md:22]: Note for Windows: I had to do: `/usr/bin/mysql` -> `./usr/bin/mysql` to get the restore command running.

**Comment by Jayracine** [comment]
Tried to restore the DB locally. It ran for a good while and then seemed to hit a problem with not enough space to write out all the contents.

```
project-swoop-mysql-1     | 2022-12-13T22:06:00.546628Z 11 [ERROR] [MY-012144] [InnoDB] posix_fallocate(): Failed to preallocate data for file ./gcdonline/gcd_story_credit.ibd, desired size 4194304 bytes. Operating system error number 28. Check that the disk is not full or a disk quota exceeded. Make sure the file system supports this function. Refer to your operating system documentation for operating system error code information.
project-swoop-mysql-1     | 2022-12-13T22:06:00.548188Z 11 [Warning] [MY-012637] [InnoDB] 1048576 bytes should have been written. Only 225280 bytes written. Retrying for the remaining bytes.
project-swoop-mysql-1     | 2022-12-13T22:06:00.549185Z 11 [Warning] [MY-012638] [InnoDB] Retry attempts for writing partial data failed.
project-swoop-mysql-1     | 2022-12-13T22:06:00.549913Z 11 [ERROR] [MY-012639] [InnoDB] Write to file ./gcdonline/gcd_story_credit.ibd failed at offset 1096810496, 1048576 bytes should have been written, only 225280 were written. Operating system error number 28. Check that your OS and file system support files of this size. Check also that the disk is not full or a disk quota exceeded.
project-swoop-mysql-1     | 2022-12-13T22:06:00.550803Z 11 [ERROR] [MY-012640] [InnoDB] Error number 28 means 'No space left on device'
```

Did we set the size of the volume for the mysql container anywhere? Are we able to increase its capacity?

File: README.md (hunk 1 above)
Comment (ajbovaird) [reply README.md:22]: I'll make this a just command and make a windows variant

**Comment by ajbovaird** [comment]
> Did we set the size of the volume for the mysql container anywhere? Are we able to increase its capacity?

The DB is pretty big uncompressed -- 7.7gb.

Is your local docker engine bumping up against a storage limit setting? (either in docker desktop or rancher desktop, whichever you're using).  Try either clearing your hanging images/volumes, or expanding the allowed storage in the settings.

**Comment by Jayracine** [comment]
I did find the setting in Rancher and the default size for a VM was set to 6GB, so that would be a problem if the DB alone surpasses that...

Pushed that up to 12GB and deleted the containers and the volume, and am starting over. It's currently chugging along at restoring the DB from the sql file.

edit: All good after expanded volume size.

**Pull request closed without merging**